Enh #6757: Allow changing visibility of global content

marc-farre opened the pull request:
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [ ] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [x] When resolving a specific issue, it's referenced in the PR's description (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [x] All tests are passing
- [ ] New/updated tests are included
- [x] Changelog was modified

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

Issue https://github.com/humhub/humhub/issues/6757

luke- reviewed (changes requested):
@marc-farre Thanks Marc. Great that it also works with the tests. :-)

Do we really need the 3 classes:
- `GlobalContentStream`
- `GlobalStreamQuery`
- `GlobalStreamFilter`

Couldn't we just integrate the condition into the existing `humhub¸\modulesdashboard\stream\filters` classes?

If we really need these classes:
- Where are they used? Expect in tests.
- Maybe we could rename them:
   - `GlobalStreamQuery` --> `GlobalContentStreamQuery`
   - `GlobalStreamFilter` --> `GlobalContentStreamFilter`

marc-farre commented:
@luke- thanks your review and suggestions.

`GlobalContentStream` relies on `GlobalContentStreamQuery` which relies on `GlobalContentStreamFilter`.

`GlobalContentStreamFilter` has 2 actions:
- Limit to global content
- Limit to public content when guest

If we remove the 3 classes, then we'll need to use directly `humhub\modules\stream\actions\Stream` to get a stream and then filter it to remove container content and private content for guests.
I think it's source of bugs and `GlobalContentStream` is easy to understand as we already have `ContentContainerStream`.
Have you got a better idea?

In the meantime, I renamed the 2 classes: https://github.com/humhub/humhub/pull/6930/commits/79d4907cb2528d3f43326021f2ff37f128a6ad28

`GlobalContentStream` is tested here: [...](https://github.com/humhub/humhub/pull/6930/files#diff-ed35209c3a9cb5d90f75fc3304ffb8528fd4746894a75930dd9fc0b9a8dd3d77R114)
As it depends on `GlobalContentStreamQuery` and `GlobalContentStreamFilter`, these 2 other classes are tested too.

However, I missed the test to limit to global content (exclude container content).
I've added `GlobalContentStreamTest::testLimitToGlobalContent()`: https://github.com/humhub/humhub/pull/6930/files#diff-ed35209c3a9cb5d90f75fc3304ffb8528fd4746894a75930dd9fc0b9a8dd3d77R100
